### PR TITLE
Ensure repositories are indexed in lexical order

### PR DESCRIPTION
**Pull request by takesson**
- Simplifies estimation of progress in multi-repo setups.
 - Simplifies detection of indexing completeness, without reading the log.

### CODE REVIEW

File: src/main/java/se/repos/indexing/standalone/IndexingDaemon.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
 	protected Injector global;
 	
 	protected Map<File, CmsRepository> known = new HashMap<File, CmsRepository>();
-	protected Map<CmsRepository, ReposIndexing> loaded = new LinkedHashMap<CmsRepository, ReposIndexing>();
+	protected Map<CmsRepository, ReposIndexing> loaded = new TreeMap<CmsRepository, ReposIndexing>(new CmsRepositoryComparator());
```
Comment (solsson) [inline src/main/java/se/repos/indexing/standalone/IndexingDaemon.java]: Maybe change this declaration from `Map` to `SortedMap`, to clarify intent?

**Comment by solsson** [comment]
Had one suggestion, but LGTM.

File: src/main/java/se/repos/indexing/standalone/IndexingDaemon.java (hunk 1 above)
Comment (takesson) [reply src/main/java/se/repos/indexing/standalone/IndexingDaemon.java]: Done

**Comment by takesson** [comment]
`Jan  9 08:43:51 ubuntu java[27803]: Exception in thread "main" java.lang.NoSuchFieldError: loaded
Jan  9 08:43:51 ubuntu java[27803]: #011at se.repos.indexing.standalone.IndexingDaemonPubSub.run(IndexingDaemonPubSub.java:153)
Jan  9 08:43:51 ubuntu java[27803]: #011at se.repos.indexing.standalone.CommandLine.runDaemonPubSub(CommandLine.java:169)
Jan  9 08:43:51 ubuntu java[27803]: #011at se.repos.indexing.standalone.CommandLine.main(CommandLine.java:97)
Jan  9 08:43:51 ubuntu systemd[1]: repos-indexing.service: Main process exited, code=exited, status=1/FAILURE
Jan  9 08:43:51 ubuntu systemd[1]: repos-indexing.service: Unit entered failed state.
Jan  9 08:43:51 ubuntu systemd[1]: repos-indexing.service: Failed with result 'exit-code'.`

**Comment by takesson** [comment]
Changing a field type of parent class requires recompile of all extending classes (even when type is "compatible"...). Requires clear workspace in Jenkins.

Surprising that Java Compiler / Jenkins still can't sort that out and there is almost no discussion about this issue, mostly confusion.

**Pull request merged**